File: Task_03/C/Hello World.c
#include <stdio.h>
#include <stdbool.h>
void PRIME(int N) {
    bool Prime[N + 1];
    for (int i = 0; i <= N; i++) {
        Prime[i] = true;
    }
    Prime[0] = Prime[1] = false;

    for (int p = 2; p * p <= N; p++) {
        if (Prime[p]) {
            for (int i = p * p; i <= N; i += p) {
                Prime[i] = false;
            }
        }
    }
    printf("Prime numbers up to %d are:\n", N);
    for (int i = 2; i <= N; i++) {
        if (Prime[i]) {
            printf("%d ", i);
        }
    }
    printf("\n");
}

int main() {
    int N;

    printf("Enter a value for N: ");
    scanf("%d", &N);

    if (N <= 1) {
        printf("N must be greater than 1.\n");
        return 1;
    }

    PRIME(N);

    return 0;
}

File: Task_03/Go/Hello World.go
package main

import (
	"fmt"
)

func Prime(n int) bool {
	if n <= 1 {
		return false
	}
	if n <= 3 {
		return true
	}
	if n%2 == 0 || n%3 == 0 {
		return false
	}
	for i := 5; float64(i)*float64(i) <= float64(n); i += 6 {
		if n%i == 0 || n%(i+2) == 0 {
			return false
		}
	}
	return true
}

func main() {
	var N int
	fmt.Print("Enter a value for N: ")
	_, err := fmt.Scanf("%d", &N)
	if err != nil || N <= 1 {
		fmt.Println("Invalid input. N must be greater than 1.")
		return
	}

	fmt.Printf("Prime numbers up to %d are:\n", N)
	for i := 2; i <= N; i++ {
		if Prime(i) {
			fmt.Printf("%d ", i)
		}
	}
	fmt.Println()
}
